File: mobile/lib/features/groups/models/group_role.dart
enum GroupRole {
  admin('admin'),
  moderator('moderator'),
  member('member'),
  guest('guest');

  const GroupRole(this.value);
  final String value;

  /// Get Vietnamese role name
  String get vietnamese {
    switch (this) {
      case GroupRole.admin:
        return 'Trưởng nhóm';
      case GroupRole.moderator:
        return 'Phó nhóm';
      case GroupRole.member:
        return 'Thành viên';
      case GroupRole.guest:
        return 'Khách';
    }
  }

  /// Get role description
  String get description {
    switch (this) {
      case GroupRole.admin:
        return 'Người tạo và quản lý nhóm, có toàn quyền';
      case GroupRole.moderator:
        return 'Hỗ trợ quản lý nhóm, có quyền hạn cao';
      case GroupRole.member:
        return 'Thành viên chính thức của nhóm';
      case GroupRole.guest:
        return 'Khách tham quan, quyền hạn hạn chế';
    }
  }

  /// Get role hierarchy level (lower number = higher authority)
  int get level {
    switch (this) {
      case GroupRole.admin:
        return 1;
      case GroupRole.moderator:
        return 2;
      case GroupRole.member:
        return 3;
      case GroupRole.guest:
        return 4;
    }
  }

  /// Check if this role can manage another role
  bool canManage(GroupRole otherRole) {
    return level < otherRole.level;
  }

  /// Get roles that can be assigned by this role
  List<GroupRole> get assignableRoles {
    switch (this) {
      case GroupRole.admin:
        return [GroupRole.moderator, GroupRole.member, GroupRole.guest];
      case GroupRole.moderator:
        return [GroupRole.member, GroupRole.guest];
      case GroupRole.member:
        return [];
      case GroupRole.guest:
        return [];
    }
  }

  /// Create from API value
  static GroupRole? fromValue(String? value) {
    if (value == null) return null;
    
    for (GroupRole role in GroupRole.values) {
      if (role.value == value) {
        return role;
      }
    }
    return null;
  }
}

/// Vietnamese-specific group roles for cultural patterns
enum VietnameseGroupRole {
  leader('truong_nhom'),
  coLeader('pho_nhom'),
  member('thanh_vien');

  const VietnameseGroupRole(this.value);
  final String value;

  /// Get Vietnamese role name
  String get vietnamese {
    switch (this) {
      case VietnameseGroupRole.leader:
        return 'Trưởng nhóm';
      case VietnameseGroupRole.coLeader:
        return 'Phó nhóm';
      case VietnameseGroupRole.member:
        return 'Thành viên';
    }
  }

  /// Get role description
  String get description {
    switch (this) {
      case VietnameseGroupRole.leader:
        return 'Người tạo và quản lý nhóm, có toàn quyền';
      case VietnameseGroupRole.coLeader:
        return 'Hỗ trợ quản lý nhóm, có quyền hạn cao';
      case VietnameseGroupRole.member:
        return 'Thành viên chính thức của nhóm';
    }
  }

  /// Get role hierarchy level (lower number = higher authority)
  int get level {
    switch (this) {
      case VietnameseGroupRole.leader:
        return 1;
      case VietnameseGroupRole.coLeader:
        return 2;
      case VietnameseGroupRole.member:
        return 3;
    }
  }

  /// Check if this role can manage another role
  bool canManage(VietnameseGroupRole otherRole) {
    return level < otherRole.level;
  }

  /// Get roles that can be assigned by this role
  List<VietnameseGroupRole> get assignableRoles {
    switch (this) {
      case VietnameseGroupRole.leader:
        return [VietnameseGroupRole.coLeader, VietnameseGroupRole.member];
      case VietnameseGroupRole.coLeader:
        return [VietnameseGroupRole.member];
      case VietnameseGroupRole.member:
        return [];
    }
  }

  /// Create from API value
  static VietnameseGroupRole? fromValue(String? value) {
    if (value == null) return null;
    
    for (VietnameseGroupRole role in VietnameseGroupRole.values) {
      if (role.value == value) {
        return role;
      }
    }
    return null;
  }
}

File: api/app/Enums/GroupRole.php
<?php

namespace App\Enums;

enum GroupRole: string
{
    case ADMIN = 'admin';           // Trưởng nhóm
    case MODERATOR = 'moderator';   // Phó nhóm
    case MEMBER = 'member';         // Thành viên
    case GUEST = 'guest';           // Khách

    /**
     * Get role name in Vietnamese
     */
    public function vietnamese(): string
    {
        return match($this) {
            self::ADMIN => 'Trưởng nhóm',
            self::MODERATOR => 'Phó nhóm',
            self::MEMBER => 'Thành viên',
            self::GUEST => 'Khách',
        };
    }

    /**
     * Get role description
     */
    public function description(): string
    {
        return match($this) {
            self::ADMIN => 'Người tạo và quản lý nhóm, có toàn quyền',
            self::MODERATOR => 'Hỗ trợ quản lý nhóm, có quyền hạn cao',
            self::MEMBER => 'Thành viên chính thức của nhóm',
            self::GUEST => 'Khách tham quan, quyền hạn hạn chế',
        };
    }

    /**
     * Get role hierarchy level (lower number = higher authority)
     */
    public function level(): int
    {
        return match($this) {
            self::ADMIN => 1,
            self::MODERATOR => 2,
            self::MEMBER => 3,
            self::GUEST => 4,
        };
    }

    /**
     * Get permissions for this role
     */
    public function permissions(): array
    {
        return match($this) {
            self::ADMIN => [
                // Full permissions - all permissions
                GroupPermission::EDIT_GROUP,
                GroupPermission::DELETE_GROUP,
                GroupPermission::CHANGE_GROUP_SETTINGS,
                GroupPermission::UPLOAD_GROUP_AVATAR,
                
                GroupPermission::ADD_MEMBERS,
                GroupPermission::REMOVE_MEMBERS,
                GroupPermission::CHANGE_MEMBER_ROLES,
                GroupPermission::VIEW_MEMBER_DETAILS,
                GroupPermission::APPROVE_MEMBER_REQUESTS,
                
                GroupPermission::CREATE_SESSIONS,
                GroupPermission::EDIT_SESSIONS,
                GroupPermission::DELETE_SESSIONS,
                GroupPermission::MANAGE_ATTENDANCE,
                
                GroupPermission::CREATE_PAYMENT_REQUESTS,
                GroupPermission::MANAGE_PAYMENTS,
                GroupPermission::VIEW_FINANCIAL_REPORTS,
                
                GroupPermission::SEND_NOTIFICATIONS,
                GroupPermission::MODERATE_DISCUSSIONS,
                
                GroupPermission::VIEW_GROUP,
                GroupPermission::JOIN_SESSIONS,
                GroupPermission::LEAVE_GROUP,
                GroupPermission::VIEW_BASIC_STATS,
            ],
            
            self::MODERATOR => [
                // Most management permissions except critical ones
                GroupPermission::EDIT_GROUP,
                GroupPermission::CHANGE_GROUP_SETTINGS,
                GroupPermission::UPLOAD_GROUP_AVATAR,
                
                GroupPermission::ADD_MEMBERS,
                GroupPermission::VIEW_MEMBER_DETAILS,
                GroupPermission::APPROVE_MEMBER_REQUESTS,
                
                GroupPermission::CREATE_SESSIONS,
                GroupPermission::EDIT_SESSIONS,
                GroupPermission::DELETE_SESSIONS,
                GroupPermission::MANAGE_ATTENDANCE,
                
                GroupPermission::CREATE_PAYMENT_REQUESTS,
                GroupPermission::VIEW_FINANCIAL_REPORTS,
                
                GroupPermission::SEND_NOTIFICATIONS,
                GroupPermission::MODERATE_DISCUSSIONS,
                
                GroupPermission::VIEW_GROUP,
                GroupPermission::JOIN_SESSIONS,
                GroupPermission::LEAVE_GROUP,
                GroupPermission::VIEW_BASIC_STATS,
            ],
            
            self::MEMBER => [
                // Basic member permissions
                GroupPermission::CREATE_SESSIONS,
                GroupPermission::EDIT_SESSIONS, // Only own sessions
                
                GroupPermission::VIEW_GROUP,
                GroupPermission::JOIN_SESSIONS,
                GroupPermission::LEAVE_GROUP,
                GroupPermission::VIEW_BASIC_STATS,
            ],
            
            self::GUEST => [
                // Very limited permissions
                GroupPermission::VIEW_GROUP,
                GroupPermission::LEAVE_GROUP,
            ],
        };
    }

    /**
     * Check if this role can manage another role
     */
    public function canManage(GroupRole $otherRole): bool
    {
        return $this->level() < $otherRole->level();
    }

    /**
     * Check if this role has a specific permission
     */
    public function hasPermission(GroupPermission $permission): bool
    {
        return in_array($permission, $this->permissions());
    }

    /**
     * Get all available roles
     */
    public static function all(): array
    {
        return [
            self::ADMIN,
            self::MODERATOR,
            self::MEMBER,
            self::GUEST,
        ];
    }

    /**
     * Get roles that can be assigned by this role
     */
    public function assignableRoles(): array
    {
        return match($this) {
            self::ADMIN => [self::MODERATOR, self::MEMBER, self::GUEST],
            self::MODERATOR => [self::MEMBER, self::GUEST],
            self::MEMBER => [],
            self::GUEST => [],
        };
    }
}